### DISCUSSION TITLE
🆕  DependabotUpdate GraphQL object Feedback

### DISCUSSION THREAD

**Discussion by exvuma** [community/community#14430] [Code Security]
We've improved the GraphQL API experience for Dependabot by adding a new `DependabotUpdate` object to the schema. We’d love to hear about how you are using it.

We designed this API to connect your security alert(s) to the Dependabot generated PR that resolves it. This can help you see when errors occur for update attempts by Dependabot and monitor Dependabot usage.

Here’s what a response body looks like: 
```json
{
 "data": {
   "repository": {
     "vulnerabilityAlerts": {
       "nodes": [
    {
            "dependabotUpdate": {
              "pullRequest": {
                "title": "build(deps): bump object-path from 0.11.5 to 0.11.8 in /npm_and_yarn/helpers",
                "number": 4772
              },
              "error": null
            }
          }
         {
            "dependabotUpdate": {
              "pullRequest": null,
              "error": {
                "body": "The latest possible version of <package> that can be installed is `1.0.6`.\n\nThe earliest fixed version is `1.0.7`.",
                "errorType": "security_update_not_possible",
                "title": "Dependabot cannot update path-parse to a non-vulnerable version"
              }
            }
          }
       ]
     }
   }
 }
}
```

Please use this discussion to submit your feedback on Dependabot’s GraphQL API and let us know what cool things you are building with it! 

For more details see the [changelog](https://github.blog/changelog/2022-04-06-dependabot-alert-api-adds-relevant-update-info-to-the-schema/).

**Comment by pawnu** [comment]
I'm trying to filter these graphql results by MERGABILITY, and SEVERITY to automate the patching of high/critical vulnerabilities. At the moment, I have to query the whole lot of alerts and manually check for these attributes which is not very efficient.

**Comment by SalimBensiali** [comment]
When Dependabot fails to create a pull request due to version incompatibility with other dependencies, I would love to more easily access info about the said blocking dependencies mentioned in the body message. Right now one needs to come up with a regex to parse the message string to extract that information.

**Comment by SalimBensiali** [comment]
@exvuma any update on this? Are you looking to further improve this part of the GraphQL object any time soon?
